pkg/csi/service/vanilla/controller.go

```diff
@@ -1487,70 +1556,72 @@ func (c *controller) createBlockVolumeWithPlacementEngineForMultiVC(ctx context.
 		},
 	}
 
-	// For topology aware provisioning, populate the topology segments parameter
-	// in the CreateVolumeResponse struct.
-	var (
-		datastoreAccessibleTopology []map[string]string
-		allNodeVMs                  []*cnsvsphere.VirtualMachine
-	)
-	// Retrieve the datastoreURL of the Provisioned Volume. If CNS CreateVolume
-	// API does not return datastoreURL, retrieve this by calling QueryVolume.
-	// Otherwise, retrieve this from PlacementResults in the response of
-	// CreateVolume API.
-	datastoreURL := volumeInfo.DatastoreURL
-	if datastoreURL == "" {
-		if volumeMgr == nil {
-			volumeMgr, err = common.GetVolumeManagerFromVCHost(ctx, c.managers, vcHost)
-			if err != nil {
-				return nil, csifault.CSIInternalFault, logger.LogNewErrorCode(log, codes.Internal, err.Error())
+	if req.AccessibilityRequirements != nil {
+		// For topology aware provisioning, populate the topology segments parameter
```

Comment:
can we move the comment before if check ?

Reply:
done.